.github/workflows/main.yml

```diff
@@ -0,0 +1,59 @@
+name: Check COSMIC Files
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+env:
+  CARGO_TERM_COLOR: always
+  # FIXME: This disables -D warnings because the build fails with it
+  RUSTFLAGS: ""
+  SKIP_TESTS: "--skip copy_dir_to_same_location \
+              --skip copy_file_to_same_location \
+              --skip copy_file_with_extension_to_same_loc \
+              --skip copy_file_with_diff_name_to_diff_dir \
+              --skip copy_to_diff_dir_doesnt_dupe_files \
+              --skip copying_files_multiple_times_to_same_location"
+
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
```

Comment:
It would probably be easier to understand if you put the strategy part first, and this line just before the line `steps`.